engine/updategraph/src/main/java/io/deephaven/engine/updategraph/UpdateGraphProcessor.java

```diff
@@ -1550,15 +1513,57 @@ private void refreshTablesAndFlushNotifications() {
         waitForNextCycle(startTime, sched);
     }
 
+    private void logCycle(final long cycleTimeNanos) {
+        if (cycleTimeNanos >= minimumCycleDurationToLogNanos) {
+            if (suppressedCycles > 0) {
+                logSuppressedCycles();
+            }
+            final long safePointPauseTimeMillis = jvmIntrospectionContext.deltaSafePointPausesTimeMillis();
+            final long safePointSyncTimeMillis = jvmIntrospectionContext.deltaSafePointSyncTimeMillis();
```

Comment:
Should these two lines go inside the `if (jvmIntrospectionContext.hasSafePointData()) {` block?